core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -41,306 +40,341 @@ object ConsumerGroupCommand {
   def main(args: Array[String]) {
     val opts = new ConsumerGroupCommandOptions(args)
 
-    if(args.length == 0)
+    if (args.length == 0)
       CommandLineUtils.printUsageAndDie(opts.parser, "List all consumer groups, describe a consumer group, or delete consumer group info.")
 
     // should have exactly one action
     val actions = Seq(opts.listOpt, opts.describeOpt, opts.deleteOpt).count(opts.options.has _)
-    if(actions != 1)
+    if (actions != 1)
       CommandLineUtils.printUsageAndDie(opts.parser, "Command must include exactly one action: --list, --describe, --delete")
 
     opts.checkArgs()
 
-    val zkUtils = ZkUtils(opts.options.valueOf(opts.zkConnectOpt), 
-                          30000,
-                          30000,
-                          JaasUtils.isZkSecurityEnabled())
+    val consumerGroupService = {
+      if (opts.options.has(opts.newConsumerOpt)) new KafkaConsumerGroupService(opts)
+      else new ZkConsumerGroupService(opts)
+    }
 
     try {
       if (opts.options.has(opts.listOpt))
-        list(zkUtils, opts)
+        consumerGroupService.list()
       else if (opts.options.has(opts.describeOpt))
-        describe(zkUtils, opts)
-      else if (opts.options.has(opts.deleteOpt))
-        delete(zkUtils, opts)
+        consumerGroupService.describe()
+      else if (opts.options.has(opts.deleteOpt)) {
+        consumerGroupService match {
+          case service: ZkConsumerGroupService => service.delete()
+          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService")
```

Comment:
I see. I guess it would be nice if we had a way to print a message that told the user that there is no need to delete group metadata for the new consumer, but not a big deal if that would require a lot of restructuring.